### PR TITLE
Add redirected-cc ninja dependency on windows builds

### PR DESCRIPTION
**Pull request by goodov**
Fix ninja dependency generation on windows for redirected *.cc files.

Resolves brave/brave-browser#14615

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: script/redirect-cc.py [hunk 1]
```diff
@@ -4,20 +4,28 @@
...
+    brave_path = replace_cc_arg(args)
     if 'CC_WRAPPER' in os.environ:
         args = [os.environ['CC_WRAPPER']] + args
     cc_retcode = subprocess.call(args)
+    # To check the redirected file timestamp, it should be marked as dependency for ninja.
+    # Linux/MacOS gcc deps format includes this file properly.
+    # Windows msvc deps format does not include it, so we do it manually here.
+    if IS_WIN32 and cc_retcode == 0 and brave_path:
+        # This is a specially crafted string that ninja will look for to create deps.
+        sys.stderr.write('Note: including file: %s\n' % brave_path)
```
Comment (goodov) [inline script/redirect-cc.py:21]: ```/showIncludes``` writes to stderr

https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2008/hdkef6tk(v=vs.90)?redirectedfrom=MSDN

**Comment by AlexeyBarabash** [comment]
@goodov , so during the compilation on Windows, ninja always uses `/showIncludes` option?

**Comment by goodov** [comment]
> @goodov , so during the compilation on Windows, does ninja always use /showIncludes option?

@AlexeyBarabash yes, gn [generates](https://gn.googlesource.com/gn/+/refs/heads/master/build/build_win.ninja.template#2) ninja files on windows to always contain this option. I can add a check to be future-proof, also it's possible to fail the build if the flag no longer exist. The check could look like this: ```'/showIncludes' in args```

**Review by AlexeyBarabash — APPROVED** [review]
++

**Review by iefremov — APPROVED** [review]

**Pull request merged**